sympy/calculus/util.py

```diff
@@ -1128,8 +1128,10 @@ def __add__(self, other):
             if other is S.Infinity and self.min is S.NegativeInfinity or \
                     other is S.NegativeInfinity and self.max is S.Infinity:
                 return AccumBounds(-oo, oo)
-            elif other.is_extended_real:
+            elif other.is_extended_real and ( (self.min is not S.NegativeInfinity) or (self.max is not S.Infinity)):
```

Comment:
Should there be `and` instead of `or`?

Reply:
it is to check if one of the limit is `oo` then this block will executes.

Reply:
If one limit is oo, then this may give wrong result.

Reply:
yes, i just saw it. i'll make the change.